src/app/shared/components/pricing-plans/pricing-plans.component.ts

```diff
@@ -96,69 +76,38 @@ export class PricingPlansComponent implements OnInit, OnChanges, OnDestroy {
       .classList.remove('active-slide');
   }
 
-  selectPlan(id) {
-    this.store.dispatch(new MembershipUpdate({ id }));
-    if (id === 1) {
-      this.selectedPlan = PlanType.PRIME;
-    } else if (id === 2) {
-      this.selectedPlan = PlanType.CHAMPION;
-    } else {
-      this.selectedPlan = PlanType.FREE;
+  setPricingPlans() {
+    if (SharedService.PRICING_PLANS_ARRAY.length > 0) {
+      this.pricingPlans = SharedService.PRICING_PLANS_ARRAY;
+      return;
     }
+    setTimeout(() => {
```

Comment:
As mentioned in other comment, if we use `Subject` for Pricing Plans then this `setTimeout` would be avoided.